src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -0,0 +1,185 @@
+
+            case nameof(UpdatePeriodicBackupCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                var backupTypes = LicenseManager.GetBackupTypes(databaseRecord.PeriodicBackups);
+
+                if (backupTypes.HasSnapshotBackup)
+                    if (serverStore.LicenseManager.LicenseStatus.HasSnapshotBackups == false)
+                        throw new LicenseLimitException(LimitType.SnapshotBackup, "Your license doesn't support adding Snapshot backups feature.");
+
+                if (backupTypes.HasCloudBackup)
+                    if (serverStore.LicenseManager.LicenseStatus.HasCloudBackups == false)
+                        throw new LicenseLimitException(LimitType.CloudBackup, "Your license doesn't support adding Cloud backups feature.");
+
+                if (backupTypes.HasEncryptedBackup)
+                    if (serverStore.LicenseManager.LicenseStatus.HasEncryptedBackups == false)
+                        throw new LicenseLimitException(LimitType.EncryptedBackup, "Your license doesn't support adding Encrypted backups feature.");
+
+                return;
+
+            case nameof(UpdatePullReplicationAsSinkCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasPullReplicationAsSink)
+                    return;
+
+                throw new LicenseLimitException(LimitType.PullReplicationAsSink, "Your license doesn't support adding Sink Replication feature.");
+
+            case nameof(UpdatePullReplicationAsHubCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasPullReplicationAsHub)
+                    return;
+
+                throw new LicenseLimitException(LimitType.PullReplicationAsHub, "Your license doesn't support adding Hub Replication feature.");
+
+            case nameof(UpdateExternalReplicationCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasExternalReplication)
+                    if (serverStore.LicenseManager.LicenseStatus.HasDelayedExternalReplication)
+                        return;
+
+                if (databaseRecord.ExternalReplications.Last().DelayReplicationFor == TimeSpan.Zero)
+                    throw new LicenseLimitException(LimitType.ExternalReplication, "Your license doesn't support adding External Replication feature.");
+
+                throw new LicenseLimitException(LimitType.DelayedExternalReplication, "Your license doesn't support adding Delayed External Replication.");
```

Comment:
You are throwing an error on both cases.

Reply:
Done